SplunkAppForWazuh/appserver/static/js/directives/wz-table/wz-table-directive.js

```diff
@@ -42,611 +42,614 @@ define([
+              )
+            }
+          }
+
+          $scope.cancelRemoveFile = () => {
+            $scope.removingFile = null
+            return init()
+          }
+
+          $scope.getWitdh = key => {
+            try {
+              if (key.includes('id') || key.includes('level')) {
+                return 'wz-width-85'
+              }
+              if (key.includes('pci') || key.includes('gdpr')) {
+                return 'wz-width-150'
+              }
+            } catch (error) { } // eslint-disable-line
+          }
+
+          /**
+           * Removes a file
+           */
+          $scope.showConfirmRemoveFile = (ev, item) => {
+            $scope.removingFile = item
+          }
+
+          $scope.isSyschecks = () => {
+            return instance.path.startsWith('/syscheck')
+          }
+
+          $scope.isWindows = () => {
+            try {
+              const agent = $scope.$parent.$parent.$parent.$parent.agent
+              return (agent.os || {}).platform === 'windows'
+            } catch (error) {
+              return false
+            }
+          }
+
+          $scope.expandItem = item => {
+            if (item.expanded) item.expanded = false
+            else {
+              $scope.pagedItems[$scope.currentPage].map(
+                item => (item.expanded = false)
+              )
+              item.expanded = true
+            }
+          }
+
+          /**
+           * Show a checkbox for each key to show or hide it
+           */
+          const cleanKeys = () => {
+            if ($scope.customColumns && sessionStorage[$scope.path]) {
+              $scope.cleanKeys = {}
+              $scope.keys.map(key => {
+                const k = key.value || key
+                let storedKeys = sessionStorage[$scope.path].split(';')
+                $scope.cleanKeys[k] = storedKeys.indexOf(k) !== -1
+              })
+            } else {
+              $scope.cleanKeys = {}
+              $scope.keys.map(key => {
+                const k = key.value || key
+                $scope.cleanKeys[k] = true
+              })
+            }
+          }
+
+          /**
+           * Launch an event to open the discover with the agent id
+           */
+          $scope.launchAgentDiscover = agentId => {
+            $scope.$emit('openDiscover', agentId)
+          }
+
+          cleanKeys()
+
+          $scope.getEquivalence = key => {
+            return $scope.keyEquivalence[key]
+          }
+
+          $scope.showCheckbox = () => {
+            $scope.showingChecks = !$scope.showingChecks
+          }
+
+          $scope.switchKey = key => {
+            $scope.cleanKeys[key] = !$scope.cleanKeys[key]
+          }
+
+          $scope.showKey = item => {
+            const it = item.value || item
+            return $scope.cleanKeys[it]
+          }
+        },
+        templateUrl:
+          BASE_URL +
+          '/static/app/SplunkAppForWazuh/js/directives/wz-table/wz-table.html'
+      }
+    })
```

Comment:
As a final note, are all the functions under `$scope` being used in the views? please make some overall searches for the method names so you can verify if they are really needed in the `$scope` or not